code-studio/src/js/levels/dialogHelper.js

```diff
@@ -157,58 +159,68 @@ window.dashboard.dialog = (function () {
    * @param {function(boolean)} onComplete Optional callback function to call when
    *     the server call completes, which receives a boolean indicating whether
    *     the browser will redirect to a new location after it is called.
+   * @param {function(function)} beforeHook Optional callback function to call
+   *     before processResults gets underway. This function must call the completion
+   *     function passed as the parameter to complete the processResults call.
    */
   // TODO(dave): move this logic into appOptions.onAttempt for levels of type
-  // external (including pixelation), multi, match, free_response, level_group,
-  // and any others which render 'levels/dialog'.
-  var processResults = function (onComplete) {
-    var results = getResult();
-    var response = results.response;
-    var result = results.result;
-    var errorType = results.errorType;
-    var testResult = results.testResult ? results.testResult : (result ? 100 : 0);
-    var submitted = results.submitted || false;
-
-
-    if (!result) {
-      showDialog(errorType || "error");
-      if (!appOptions.dialog.skipSound) {
-        CDOSounds.play('failure');
-      }
+  // external (including pixelation), multi, match, and any others
```

Comment:
Did you mean to change this comment? free_response and level_group still render levels/dialog.